pkg/cli/alpha/scaffold.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpha
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+)
+
+// NewScaffoldCommand return a new scaffold command
+func NewScaffoldCommand() *cobra.Command {
+	var projectConfig, outputPath string
+	scaffoldCmd := &cobra.Command{
+		Use:   "scaffold",
+		Short: "Re-scaffold an existing kuberbuilder project",
+		Long: `This command is a helper for you upgrade your project to the latest versions scaffold.
+		
+		It will:
+			- Create a new directory name after the project in the current working directory or at the output path
+			- Re-generate the whole project based on the Project file data
+
+		You will still need to move your code and other customizations over.`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return validate(projectConfig, outputPath)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			projectPath := getProjectPath(cwd, projectConfig)
```

Comment:
We need to copy the current project to an old-<project> name dir so that the users can compare both and add the code on top of the project that was re-scaffold

Reply:
this would create a subdirectory with a re-scaffolded project, why would we need to touch the old project?

Reply:
As a user, we create the project and add code on top.
So that to migrate the project we need to re-scaffold all and then compare the "old project" with the new scaffold so that we can re-add all code done on top again. 

So, ihmo we need to do here: 

a) copy the current project to another dir ( i.e. cp -R * my-operator/ old-my-operator/ )
b) then, in the current dir we will remove all files rm -rf *
c) after that we can re-scaffold all so that a user can use git OR compare my-operator/ with old-my-operator/ in order to add the code again.